### PR TITLE
Override settings per-request

### LINKED ISSUE #5: Can't POST to some sub-resources
```
"campaigns.schedule.create": {
      "method": "POST",
      "url": "emailmarketing/campaigns/**:campaignId**/schedules"
    }
```
In this case, I need to be able to POST `{ }` to the endpoint, which is a sub-resource of a specific campaign. The only way to populate the _campaignId_ param is to pass it in the params object, which is then included in the request body. This is not the desired behaviour, as I can't independently fill in the params and control the request body.

### PR DESCRIPTION
**Pull request by szb0018**
Resolves #5

### CODE REVIEW

File: lib/resource.js [hunk 1]
```diff
@@ -23,7 +23,7 @@ function Resource(clientFactory, actions) {
         url = url.param(params, true).query(qs).toString();
         debug('resource')(`HTTP ${config.method} ${url}`);
         debug('resource')('body:', params);
-        var req = client[method](url).body(params);
+        var req = client[method](url).body(data || params);
```
Comment (compwright) [inline lib/resource.js:26]: Should we merge these objects together, instead of substituting? Is there a scenario where you need to retain certain params set, while customizing others per-request?

**Comment by szb0018** [comment]
Yeah, I put more detailed notes in the issue I linked the PR to.

tl;dr I need to fill in a path param and send a body that does not include the param.

Get Outlook for iOS<https://aka.ms/o0ukef>
________________________________

**Comment by compwright** [comment]
I'll wait to merge this then. Object.assign() will do the trick, as long as we don't have to worry about nested props.

**Comment by szb0018** [comment]
What are you waiting on? Object.assign will merge the props, which is what I am trying to avoid.

**Comment by compwright** [comment]
Oh okay I see what you mean - went back and read the related issue and it makes sense now.
